### DISCUSSION TITLE
Late recording of particle positions?

### DISCUSSION THREAD

**Discussion by AndresSepulveda** [OpenDrift/opendrift#811] [Q&A]
Quick question. Is there a way to record the particle position after a certain moment? Say the last 20 days/time steps of a simulation....

The idea is that if a larvae can settle  after a growth period, so I am only interested in its position after it reached the given maturity, in this case, calculated by time only.

**Comment by knutfrode** [comment]
`self.history`contains the full history (properties and environment/forcing) of all particles.
Though, for simulations with many time steps and particles this is flushed to disk if too much to keep in memory.
You can control this frequency with input parameter `export_buffer_length` to `run()` (default 100 output time steps)

**Reply by camines** [reply]
Hello, 

About the self.history, I'm trying to see the average age of the particles, but I don't know how to access to the data(I guess it should be stored as a list, or an array). How can I call the self.history?

It is possible to access directly after the run to this information, and have as well i.e. the position of a particle that is stranded or deactivated? Or for doing so I need to extract a NetCDF of the run, and read it somewhere else?

Thanks for your time and help :)

**Reply by knutfrode** [reply]
During the simulation (within your module), the age is available as `self.elements.age_seconds`  (array with one value per active element)

After the simulation, age (and other properties) are availabe in the record array `o.history`  (or `self.history`) with dimensions (time_step, element_number), as: `age = o.history['age_seconds']`

You can also get any property through a convenience method:
`age, status = o.get_property('age_seconds')`

where age is the property, and status is 0 for active particles and >0 for deactivated particles (e.g. stranded)

**Comment by AndresSepulveda** [comment]
For what I understand this only would change how often I write the output file.

 Is there a way not to write the first X output time steps?

**Comment by knutfrode** [comment]
There is presently no way achieve this. Though you can truncate the output netCDF file after the run with tools like `Nco` or `CDO`
Also, it is possible to import only a subset of the file in time like:
`o = opendrift.open(<netCDF-file>, times=range(20, 50))`
to import only from time step 20 until time step 50.
